api/v1alpha3/virtualmachine_types.go

```diff
@@ -161,6 +161,17 @@ const (
 	// PVCDiskDataExtraConfigKey is the ExtraConfig key to persist the VM's
 	// PVC disk data in JSON, compressed using gzip and base64-encoded.
 	PVCDiskDataExtraConfigKey = "vmservice.virtualmachine.pvc.disk.data"
+
+	// DisableAutoRegistrationExtraConfigKey is the ExtraConfig key that can be
+	// set to "true" (case insensitive) on a virtual machine to opt-out of the
+	// automatic registration workflow.
+	//
+	// After a restore from a backup restore vendor, or a failover from a disaster
+	// recovery solution, vCenter automatically attempts to register the restored
+	// virtual machine with Supervisor. If the automatic registration is skipped by
+	// specifying this key, backup/restore and/or disaster recovery solutions are
+	// responsible to register the VM with Supervisor.
+	DisableAutoRegistrationExtraConfigKey = "vmservice.virtualmachine.disable-automatic-registration"
```

Comment:
Not sure if there is any convention on the EC key name, but most of the existing ones use "." as a separator.

Reply:
The "." separator is used between things that provide a hierarchy (e.g., "vmservice", and "virtualmachine").  As far as the "disable-automatic-registration" is concerned, I could either use camel case of snake case.  I am seeing references for both in the existing EC keys.  Going to use camel case since that's what is more prevalent.